simulate/simulate.go

```diff
@@ -335,22 +336,36 @@ func BasesToBaseExt(seq []dna.Base) []BaseExt {
 
 //BaseExtToBases converts a slice of BaseExt to a slice of dna.Base
 func BaseExtToBases(seq []BaseExt) []dna.Base {
-	var newSequence []dna.Base
-	var i, j int
+	var newSequence []dna.Base = make([]dna.Base, len(seq))
 
-	for i = 0; i < len(seq); i++ { //counter to put things back in order
-		for j = 0; j < len(seq); j++ { //counter to check all seq bases before incrementing to look for the next base in the newSeq
-			if seq[j].SeqPos == i {
-				newSequence = append(newSequence, seq[j].Base)
-			}
-		}
+	SortBaseExtBySeqPos(seq)
+	for i := 0; i < len(seq); i++ {
+		newSequence[i] = seq[i].Base
 	}
 	if len(newSequence) != len(seq) {
 		log.Fatal("Cannot find order of bases")
 	}
 	return newSequence
 }
 
+//SortBaseExtBySeqPos orders a string of BaseExt by seq position
+func SortBaseExtBySeqPos(unordered []BaseExt) {
+	sort.Slice(unordered, func(i, j int) bool {
+		return Compare(unordered[i], unordered[j]) == 0
```

Comment:
Should the check here be `Compare(...) < 0` instead of `Compare(...) == 0`?

Reply:
I'm doing a test to check this is actually functioning, but I'm not totally sure, I haven't really fully grasped how this function works yet.